notify-debouncer-full/src/cache.rs

```diff
@@ -38,57 +34,68 @@ pub trait FileIdCache {
     }
 }
 
-/// A cache to hold the file system IDs of all watched files.
-///
-/// The file ID cache uses unique file IDs provided by the file system and is used to stich together
-/// rename events in case the notification back-end doesn't emit rename cookies.
-#[derive(Debug, Clone, Default)]
-pub struct FileIdMap {
-    paths: HashMap<PathBuf, FileId>,
-}
-
-impl FileIdMap {
-    /// Construct an empty cache.
-    pub fn new() -> Self {
-        Default::default()
+#[cfg(not(any(target_os = "linux", target_os = "android", target_family = "wasm")))]
+mod file_id_map {
```

Comment:
I would prefer this to be its own file. No need to make it a sub-module of `cache`, `crate::file_id_map` should be fine.